nixio/cmd/nixexplore.py

```diff
@@ -0,0 +1,622 @@
+    for e in entities:
+        if isinstance(e, nix.DataArray):
+            dump_data_array(e, filename)
+
+    nix_file.close()
+
+
+def data_plotter(filename, arguments):
+    nix_file = open_nix_file(filename)
+    entities = find_data_entity(nix_file, arguments)
+    for e in entities:
+        if isinstance(e, nix.DataArray):
+            plotter = nw.plotter.suggested_plotter(e)
+            if plotter:
+                plotter.plot()
+                plt.show()
+            else:
+                print("Could not find a suitable plotter for the DataArray: %s" % str(e))
+        else:
+            print("Sorry, so far I can only try to plot DataArrays.")
+
+    nix_file.close()
+
+
+def disp_data(filename, arguments):
+    nix_file = open_nix_file(filename)
+    entities = find_data_entity(nix_file, arguments)
+    print("# File: %s" % filename)
+    for e in entities:
+        print("# entity: %s\n# type: %s\n# id: %s" % (e.name, e.type, e.id))
+        print("# created at: %s\n# last edited at: %s\n" %
+              (str(dt.datetime.fromtimestamp(e.created_at)),
+               str(dt.datetime.fromtimestamp(e.updated_at))))
+        print(e)
+        print("\n")
+    nix_file.close()
+
+
+def data_worker(arguments):
+    files = assemble_files(arguments)
+    for nf in files:
+        disp_data(nf, arguments)
+
+
+def file_worker(arguments):
+    files = assemble_files(arguments)
+    for nf in files:
+        disp_file_info(nf, arguments)
+
+
+def dump_worker(arguments):
+    files = assemble_files(arguments)
+    func = data_dump
+    for nf in files:
+        func(nf, arguments)
+
+
+def plot_worker(arguments):
+    files = assemble_files(arguments)
+    func = data_plotter
+    for nf in files:
+        func(nf, arguments)
+
+
+def add_default_file_args(parent_parser):
+    parent_parser.add_argument("file", type=str, nargs="+",
+                               help="Path to file (at least one)")
+    parent_parser.add_argument("-s", "--suffix", type=str, default="nix", nargs="?",
+                               help="The file suffix used for nix data files (default: %(default)s).")
+
+
+def add_default_args(parent_parser):
+    parent_parser.add_argument("-c", "--case_sensitive", action="store_true", help="matching of"
+                               + " entitiy names and types is case sensitive, by default the case is ignored")
+    parent_parser.add_argument("-fm", "--full_match", action="store_true", help="names and types must"
+                               + " be full matches, bey default a partial match is sufficient")
+
+
+def create_metadata_parser(parent_parser):
+    meta_parser = parent_parser.add_parser("metadata", help="Filter and display metadata", aliases=["mdata"],
+                                           description="Search for metadata items or display metadata (sub)trees.")
+    meta_parser.add_argument("-p", "--pattern", type=str, default=[], nargs="+", help=mdata_pattern_help)
```

Comment:
The `nargs="+"` here creates an issue where running something like:
```
nixio-explore metadata -p "pattern" <filename>
```
causes `<filename>` to be interpreted as a second pattern and then the command fails with
```
nixio-explore metadata: error: the following arguments are required: file
```

The user can get around this by instead calling
```
nixio-explore metadata -p "pattern" -- <filename>
```
but this isn't immediately obvious and we should find a better way to handle multiple patterns.

Reply:
using the "append" method now so multiple patterns are passed like `nixio-explore metadata -p pattern1 -p pattern2 file1 file2 file3`